parquet/src/main/java/org/apache/iceberg/parquet/ParquetFileMerger.java

```diff
@@ -0,0 +1,421 @@
+
+          for (BlockMetaData rowGroup : rowGroups) {
+            long rowCount = rowGroup.getRowCount();
+            writer.startBlock(rowCount);
+
+            // Copy all existing column chunks (binary copy)
+            copyColumnChunks(writer, baseSchema, inputFile, rowGroup);
+
+            // Write new _row_id column chunk
+            writeRowIdColumnChunk(writer, rowIdDescriptor, currentRowId, rowCount);
+            currentRowId += rowCount;
+            writer.endBlock();
+          }
+        }
+      }
+
+      if (extraMetadata != null && !extraMetadata.isEmpty()) {
+        writer.end(extraMetadata);
+      } else {
+        writer.end(emptyMap());
+      }
+    }
+  }
+
+  /**
+   * Merges multiple Parquet files with optional row lineage preservation.
+   *
+   * <p>This method intelligently handles row lineage based on the input files and firstRowIds:
+   *
+   * <ul>
+   *   <li>If firstRowIds is null/empty: performs simple binary copy merge
+   *   <li>If files already have physical _row_id column: performs simple binary copy merge
+   *   <li>Otherwise: synthesizes physical _row_id column from virtual metadata
+   * </ul>
+   *
+   * @param inputFiles List of Iceberg input files to merge
+   * @param encryptedOutputFile Encrypted output file for the merged result (encryption handled
+   *     based on table configuration)
+   * @param schema Parquet schema from the input files (assumed already validated)
+   * @param firstRowIds Optional list of starting row IDs for each input file (null if no lineage
+   *     needed)
+   * @param rowGroupSize Target row group size in bytes
+   * @param columnIndexTruncateLength Maximum length for min/max values in column index
+   * @param extraMetadata Additional metadata to include in the output file footer (can be null)
+   * @return true if output file has physical _row_id column, false otherwise
+   * @throws IOException if I/O error occurs during merge operation
+   */
+  public static boolean mergeFilesWithOptionalRowIds(
+      List<InputFile> inputFiles,
+      EncryptedOutputFile encryptedOutputFile,
+      MessageType schema,
+      List<Long> firstRowIds,
+      long rowGroupSize,
+      int columnIndexTruncateLength,
+      Map<String, String> extraMetadata)
+      throws IOException {
+    // Get the encrypting output file (encryption applied based on table configuration)
+    OutputFile outputFile = encryptedOutputFile.encryptingOutputFile();
+
+    // Check if row lineage processing is requested
+    boolean needsRowLineageProcessing = firstRowIds != null && !firstRowIds.isEmpty();
+
+    if (needsRowLineageProcessing) {
+      if (schema.containsField(MetadataColumns.ROW_ID.name())) {
+        // Files already have physical _row_id - use simple binary copy (fastest!)
+        mergeFilesWithSchema(
+            inputFiles, outputFile, schema, rowGroupSize, columnIndexTruncateLength, extraMetadata);
+        return true; // Output has physical _row_id from input
```

Comment:
The return value is not used anywhere. Can we return the value if the merge was successful or not? In this case we don't need to expose `canMerge` in a different method.